### PR TITLE
DocumentSelectTaskSerializer now includes document type

### PR DESCRIPTION
**Pull request by damm89**
Dependent on https://github.com/GemeenteUtrecht/ZGW/pull/738

### CODE REVIEW

File: backend/src/zac/core/camunda/select_documents/serializers.py [hunk 1]
```diff
@@ -53,14 +53,52 @@ class Meta:
 #
 
 
+class SelectedDocumentSerializer(serializers.Serializer):
+    document = serializers.ChoiceField(
+        child=serializers.URLField,
```
Comment (damm89) [inline backend/src/zac/core/camunda/select_documents/serializers.py:58]:
```suggestion
```

File: backend/src/zac/core/camunda/select_documents/serializers.py [hunk 2]
```diff
@@ -53,14 +53,51 @@ class Meta:
...
 
+class SelectedDocumentSerializer(serializers.Serializer):
+    document = serializers.ChoiceField(
+        label=_("Selected document"),
+        help_text=_("The URL of the selected document from the relevant case."),
+        choices=(),
+    )
+
+    document_type = serializers.URLField(
+        child=serializers.URLField,
```
Comment (damm89) [inline backend/src/zac/core/camunda/select_documents/serializers.py:64]:
```suggestion
    document_type = serializers.ChoiceField(
```

File: backend/src/zac/core/api/views.py [hunk 3]
```diff
@@ -456,7 +456,6 @@ def post(self, request: Request) -> Response:
 class InformatieObjectTypeListView(generics.ListAPIView):
     """
     List the available document types for a given zaak.
-
```
Comment (sergei-maertens) [inline backend/src/zac/core/api/views.py:459]: PEP 257 asks you to format docstrings like this:

```
A short summary, preferably < 70-ish characters.

A more elaborate description, following a blank line.
```

This has an impact on how editors, documentation-generation tools etc. show class/function documentation.

File: backend/src/zac/core/camunda/select_documents/serializers.py [hunk 4]
```diff
@@ -53,27 +55,107 @@ class Meta:
...
+        self.zaak = self.get_zaak_from_context()
+
+        # Validate selected documents
+        doc_urls = [doc["document"] for doc in selected_docs]
+        validate_zaak_documents(doc_urls, self.zaak)
+
+        # Validated selected document types according to case type
+        ztc_client = _client_from_url(self.zaak.zaaktype)
+        results = get_paginated_results(ztc_client, "informatieobjecttype")
+        valid_eiots = [iot["url"] for iot in results]
```
Comment (sergei-maertens) [inline backend/src/zac/core/camunda/select_documents/serializers.py:92]: This is not the set restricted to the zaaktype - for that you need to check `zaaktype.informatieobjecttypen` (see https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/VNG-Realisatie/catalogi-api/1.0.0/src/openapi.yaml#operation/zaaktype_read)

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: backend/src/zac/core/api/views.py (hunk 3 above)
Comment (damm89) [reply backend/src/zac/core/api/views.py:459]: Accidentally edited this!

File: backend/src/zac/core/camunda/select_documents/serializers.py (hunk 4 above)
Comment (damm89) [reply backend/src/zac/core/camunda/select_documents/serializers.py]: Good catch

**Review by sergei-maertens — APPROVED** [review]
I'm expecting some rework once Kelvin gets started with this, since it's not clear for him now which informatieobjecttypen should be shown in the UI and which don't.

https://zac-test.utrechtproeftuin.nl/api/docs/#operation/core_document_types_list is the closest we have, but not sufficient, because we should support `zaaktype` querystring here as well, and pass `zaaktype` to the frontend so it can be used to query.

**Pull request merged**